addons/mail/wizard/mail_compose_message.py

```diff
@@ -808,7 +810,8 @@ def _prepare_mail_values(self, res_ids):
 
         if email_mode:
             mail_values_all = self._process_mail_values_state(mail_values_all)
-        return mail_values_all
+        return (mail_values_all if not email_mode or mass_include_canceled
+                else {res_id: values for res_id, values in mail_values_all.items() if values.get('state') != 'cancel'})
```

Comment:
I think it might be best to split this in a real if-else block actually. Might be a bit more readable now, and especially if we want to add conditions later. Which I might possibly want for my task actually ^^